### PR TITLE
Removing provisioner's dependency on csi types

### PR DESCRIPTION
**Pull request by mukhoakash**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/apis/azuredisk/v1alpha1/types.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"github.com/golang/protobuf/ptypes/timestamp"
```
Comment (edreed) [inline pkg/apis/azuredisk/v1alpha1/types.go:20]: I don't believe we can take a dependency on  the protobuf `Timestamp` in the CRDs. We should probably use [`metav1.Time`] which is the K8s native API type for representing timestamps.

File: pkg/azureutils/azure_snapshot_utils.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func GetSnapshotAndResourceNameFromSnapshotID(snapshotID string) (snapshotName,
 	return snapshotName, resourceGroup, err
 }
 
-func GenerateCSISnapshot(sourceVolumeID string, snapshot *compute.Snapshot) (*csi.Snapshot, error) {
+func GenerateV1Alpha1Snapshot(sourceVolumeID string, snapshot *compute.Snapshot) (*v1alpha1.Snapshot, error) {
```
Comment (edreed) [inline pkg/azureutils/azure_snapshot_utils.go:40]: Consider `NewAzureDiskSnapshot`. We will eventually change the API version used internally: we should avoid using versions in our function names.

File: pkg/apis/azuredisk/v1alpha1/types.go (hunk 1 above)
Comment (edreed) [inline pkg/apis/azuredisk/v1alpha1/types.go:20]: I don't believe we can take a dependency on  the protobuf `Timestamp` in the CRDs. We should probably use [`metav1.Time`] which is the K8s native API type for representing timestamps.

**Review by edreed — CHANGES_REQUESTED** [review]
CRD config, deployment YAML files and Helm charts need to be updated.

File: pkg/apis/azuredisk/v1alpha1/types.go (hunk 1 above)
Comment (mukhoakash) [reply pkg/apis/azuredisk/v1alpha1/types.go:20]: I might be able to use int64 similar to how its been used by `LastHeartbeatTime` in the `AzDriverNode` object

Comment (edreed) [reply pkg/apis/azuredisk/v1alpha1/types.go:20]: I'd rather use the standard types for timestamps. `LastHeartbeatTime` should probably be a `metav1.Time` as well.

**Comment by edreed** [comment]
The e2e test appear to fail in `AfterSuite` due to an error unrelated to your changes. I noticed this last night in my PRs. It appears to be related to [kubernetes-sigs/azuredisk-csi-driver#836](https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/836) and requires K8s 1.18 or later to run. The [azure-cluster-up.sh](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/4284765381d28a3143aeb1e097c399c11fd3bd8d/hack/azure-cluster-up.sh#L252) uses 1.17 by default. I'll update this today.

```console
2021/05/26 14:10:54 ===================install Azure Disk CSI Driver deployment scripts test===================
use local deploy
Installing Azure Disk CSI driver, version: master ...
error: unable to recognize "./deploy/csi-azuredisk-driver.yaml": no matches for kind "CSIDriver" in version "storage.k8s.io/v1"

------------------------------
Failure [219.170 seconds]
[AfterSuite] AfterSuite 
/home/runner/work/azuredisk-csi-driver/azuredisk-csi-driver/test/e2e/suite_test.go:124

  Unexpected error:
      <*exec.ExitError | 0xc000660080>: {
          ProcessState: {
              pid: 61739,
              status: 256,
              rusage: {
                  Utime: {Sec: 0, Usec: 316068},
                  Stime: {Sec: 0, Usec: 65236},
                  Maxrss: 120516,
                  Ixrss: 0,
                  Idrss: 0,
                  Isrss: 0,
                  Minflt: 3061,
                  Majflt: 0,
                  Nswap: 0,
                  Inblock: 16,
                  Oublock: 13312,
                  Msgsnd: 0,
                  Msgrcv: 0,
                  Nsignals: 0,
                  Nvcsw: 1922,
                  Nivcsw: 147,
              },
          },
          Stderr: nil,
      }
      exit status 1
  occurred

  /home/runner/work/azuredisk-csi-driver/azuredisk-csi-driver/test/e2e/suite_test.go:252
------------------------------

Failed to create JUnit directory: /workspace/_artifacts/junit_01.xml
	mkdir /workspace: permission deniedFailed to create JUnit report file: /workspace/_artifacts/junit_01.xml
	open /workspace/_artifacts/junit_01.xml: no such file or directory
Failed to generate JUnit report data:
	invalid argument
Ran 54 of 85 Specs in 7641.473 seconds
FAIL! -- 54 Passed | 0 Failed | 0 Pending | 31 Skipped
--- FAIL: TestE2E (7641.49s)
FAIL
FAIL	sigs.k8s.io/azuredisk-csi-driver/test/e2e	7641.541s
FAIL
make: *** [Makefile:344: e2e-test-v2] Error 1
Error: Process completed with exit code 2.
```

**Review by edreed — APPROVED** [review]

**Pull request merged**